lib/widgets/all_channels.dart

```diff
@@ -96,28 +98,30 @@ class AllChannelsListEntry extends StatelessWidget {
     final Subscription? subscription = channel is Subscription ? channel : null;
     final hasContentAccess = store.selfHasContentAccess(channel);
 
-    return Padding(
-      padding: EdgeInsetsDirectional.only(start: 8, end: 4),
-      child: Row(spacing: 6, children: [
-        Icon(
-          size: 20,
-          color: colorSwatchFor(context, subscription).iconOnPlainBackground,
-          iconDataForStream(channel)),
-        Expanded(
-          child: Text(
-            style: TextStyle(
-              color: designVariables.textMessage,
-              fontSize: 17,
-              height: 20 / 17,
-            ).merge(weightVariableTextStyle(context, wght: 600)),
-            channel.name)),
-        if (hasContentAccess) _SubscribeToggle(channel: channel),
-        ZulipIconButton(
-          icon: ZulipIcons.more_horizontal,
-          onPressed: () {
-            showChannelActionSheet(context, channelId: channel.streamId);
-          }),
-      ]));
+    return InkWell(
+      onTap: !hasContentAccess ? null : () => Navigator.push(context,
+        MessageListPage.buildRoute(context: context,
+          narrow: ChannelNarrow(channel.streamId))),
+      onLongPress: () => showChannelActionSheet(context, channelId: channel.streamId),
+      child: Padding(
+        padding: const EdgeInsets.fromLTRB(8, 2, 12, 2),
+        child: Row(spacing: 6, children: [
+          Icon(
+            size: 20,
+            color: colorSwatchFor(context, subscription).iconOnPlainBackground,
+            iconDataForStream(channel)),
+          Expanded(
+            child: Text(
+              maxLines: 1,
+              overflow: TextOverflow.ellipsis,
+              style: TextStyle(
+                color: designVariables.textMessage,
+                fontSize: 17,
+                height: 20 / 17,
+              ).merge(weightVariableTextStyle(context, wght: 600)),
+              channel.name)),
+          hasContentAccess ? _SubscribeToggle(channel: channel) : const SizedBox(height: 40, width: 52)
```

Comment:
Add SizedBox(height: 40, width: 52), the size of the toggle, as a
    placeholder for hidden toggle to prevent UI jump. Adjust Row padding

This seems pretty fragile.

Is the toggle always 40px high, or might it sometimes be taller or shorter? It's not obvious to me that it's definitely always the same height.

Even if it's currently always that height, will it always be that height in the future? That doesn't sound like something I'd expect it to guarantee.